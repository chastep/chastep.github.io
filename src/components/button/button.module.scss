@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.button {
  border: 1px solid hsl(var(--color-border));
  border-radius: variables.$button-border-radius;
  font-size: variables.$typography-base-font-size;
  line-height: variables.$button-height;
  color: hsl(var(--color-foreground));
  height: variables.$button-height;
  text-decoration: none;
  display: inline-block;
  font-weight: normal;
  margin-right: auto;
  text-align: center;
  margin-left: auto;
  @include mixins.padding(0, 1);

  &:hover, &:focus {
    color: hsl(var(--color-accent-foreground));
    background: hsl(var(--color-accent));
  }
}
