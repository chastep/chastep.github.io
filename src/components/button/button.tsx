import React, { type FC } from "react";

import cn from "classnames";
import { Link } from "gatsby";

import * as styles from "./button.module.scss";

interface ButtonProps {
  className?: string;
  title: string;
  to: string;
}

const Button: FC<ButtonProps> = ({ className, title, to }) => (
  <Link className={cn(styles.button, className)} to={to}>
    {title}
  </Link>
);

export { Button };
