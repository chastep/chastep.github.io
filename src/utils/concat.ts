const concat = (...args: string[]): string => args.join("");

export { concat };
