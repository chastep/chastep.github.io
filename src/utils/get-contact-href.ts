const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    mastodon: `${contact}`,
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    x: `https://www.x.com/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    bluesky: `https://bsky.app/profile/${contact}.bsky.social`,
  };

  return hrefs[name] ?? contact;
};

export { getContactHref };
