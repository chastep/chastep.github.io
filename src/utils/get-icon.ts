import { icons } from "@/constants/icons";

const getIcon = (name: keyof typeof icons) => icons[name] || {};

export { getIcon };
