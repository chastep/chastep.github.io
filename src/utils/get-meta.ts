const getMeta = (container: HTMLElement, key: string) => {
  const meta = container.getElementsByTagName("meta");

  for (let i = 0; i < meta.length; i += 1) {
    if (
      key === meta[i].getAttribute("name") ||
      key === meta[i].getAttribute("property")
    ) {
      return meta[i].getAttribute("content");
    }
  }

  return "";
};

export { getMeta };
