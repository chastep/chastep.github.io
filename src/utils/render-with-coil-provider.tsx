import React, { type ReactElement, type ReactNode, type FC } from "react";
import { CoilProvider } from "@alxshelepenok/diesel";
import { render, type RenderOptions } from "@testing-library/react";
import renderer, { type ReactTestRenderer, type TestRendererOptions } from "react-test-renderer";

interface WithCoilProviderProps {
  children: ReactNode;
}

const WithCoilProvider: FC<WithCoilProviderProps> = ({ children }) => (
  <CoilProvider>{children}</CoilProvider>
);

const renderWithCoilProvider = (
  ui: ReactElement,
  options?: RenderOptions,
) => render(ui, { wrapper: WithCoilProvider, ...options });

export const createSnapshotsRenderer = (
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer =>
  renderer.create(<CoilProvider>{nextElement}</CoilProvider>, options);

export { renderWithCoilProvider };
