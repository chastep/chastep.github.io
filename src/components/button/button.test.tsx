import React from "react";
import { describe, test, expect } from "bun:test";
import { render as reactTestingLibraryRender } from "@testing-library/react";

import { Button } from "@/components/button/button";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("Button", () => {
  test("renders correctly", () => {
    const props = { title: "Button", to: "/" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with external link", () => {
    const props = { title: "Button", to: "https://example.com" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with custom class name", () => {
    const props = { title: "Button", to: "/", className: "custom-class-name" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("the title is rendered correctly and exists", () => {
    const props = { title: "Text", to: "/" };
    const tree = reactTestingLibraryRender(<Button {...props} />);
    expect(tree.getByText(props.title)).toBeInTheDocument();
  });
});
