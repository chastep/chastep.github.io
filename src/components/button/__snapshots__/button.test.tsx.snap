// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "Button",
  ],
  "props": {
    "aria-current": "page",
    "className": "",
    "href": "/",
    "onClick": [Function],
    "onMouseEnter": [Function],
    "style": {},
  },
  "type": "a",
}
`;

exports[`Button renders correctly with external link 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "Button",
  ],
  "props": {
    "className": "",
    "href": "https://example.com",
  },
  "type": "a",
}
`;

exports[`Button renders correctly with custom class name 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "Button",
  ],
  "props": {
    "aria-current": "page",
    "className": "custom-class-name",
    "href": "/",
    "onClick": [Function],
    "onMouseEnter": [Function],
    "style": {},
  },
  "type": "a",
}
`;
